components/engine_test/src/lib.rs

```diff
@@ -267,27 +257,22 @@ pub mod kv {
 
             let mut reg = self.registry.lock().unwrap();
             if let Some(suffix) = suffix {
-                if let Some(tablet) = reg.get(&(id, suffix)) {
+                if let Some((cached_tablet, cached_suffix)) = reg.get(&id) && *cached_suffix == suffix {
                     // Target tablet exist in the cache
-
                     if options.create_new() {
-                        return Err(box_err!("region {} {} already exists", id, tablet.path()));
+                        return Err(box_err!("region {} {} already exists", id, cached_tablet.path()));
                     }
-                    return Ok(tablet.clone());
+                    return Ok(cached_tablet.clone());
                 } else if !options.cache_only() {
                     let tablet_path = self.tablet_path(id, suffix);
                     let tablet = self.open_tablet_raw(&tablet_path, id, suffix, options.clone())?;
                     if !options.skip_cache() {
-                        reg.insert((id, suffix), tablet.clone());
+                        reg.insert(id, (tablet.clone(), suffix));
                     }
                     return Ok(tablet);
                 }
-            } else if options.cache_only() {
-                // This branch reads an arbitrary tablet with region id `id`
-
-                if let Some(k) = reg.keys().find(|k| k.0 == id) {
-                    return Ok(reg.get(k).unwrap().clone());
-                }
+            } else if let Some((tablet, _)) = reg.get(&id) {
```

Comment:
What if `create_new` is set to true?

Reply:
If create_new is set be true, then suffix should not be None.

Reply:
How to enforce that?

Reply:
I can add a check and return an err in that case.